ibs/chugs/views.py

```diff
@@ -1,3 +1,143 @@
-from django.shortcuts import render
+from rest_framework.decorators import api_view, permission_classes, parser_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.parsers import JSONParser
 
-# Create your views here.
+from django.db import transaction
+from django.db.models import Count
+from django.db.models import F
+
+from ibs.users.models import User
+
+from ibs.chugs.models import Strafbak
+from ibs.chugs.serializers import StrafbakSerializer
+
+@api_view(['GET', 'POST', 'DELETE'])
+@permission_classes([IsAuthenticated])
+@parser_classes([JSONParser])
+def strafbakken(request):
+  match request.method:
+    case 'GET':
+      return get_strafbakken()
+    case 'POST':
+      return give_strafbak(request)
+    case 'DELETE':
+      return trek_strafbak(request)
+  return Response(status=status.HTTP_400_BAD_REQUEST)
+
+# Get details about your strafbakken
+def get_strafbakken():
+  try:
+    strafbakkenCount = Strafbak.objects.filter(active=True).annotate(name=F('receiver__username')).values('name').annotate(bakken=Count('receiver'))
+    return Response(list(strafbakkenCount), status=status.HTTP_200_OK)
+  except Exception as error:
+    print(error)
+    return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+# Give a strafbak
+def give_strafbak(request):
+  request.data['giver'] = request.user.id
+
+  try:
+    users = User.objects.filter(username=request.data['receiver'])
+  except Exception as error:
+    print(error)
+    return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+  if (len(users) == 0):
+      return Response(status=status.HTTP_400_BAD_REQUEST)
+  request.data['receiver'] = users[0].id
+
+  serializer = StrafbakSerializer(data=request.data)
+  if serializer.is_valid():
+    try:
```

Comment:
als de conditie serializer.is_valid() True is dan gooit het sws geen exception, dus de try except kan weg